src/main/java/org/prebid/server/bidder/BidderCatalog.java

```diff
@@ -13,10 +14,17 @@
  */
 public class BidderCatalog {
 
+    private static final String ERROR_MESSAGE_TEMPLATE_FOR_DEPRECATED = "%s has been deprecated and is no "
+            + "longer available. Use %s instead.";
+
     private final Map<String, BidderDeps> bidderDepsMap;
 
+    private final Map<String, String> deprecatedNameToError = new HashMap<>();
+
     public BidderCatalog(List<BidderDeps> bidderDeps) {
         bidderDepsMap = Objects.requireNonNull(bidderDeps).stream()
+                .peek(deps -> deprecatedNameToError.putAll(
```

Comment:
Please either pull this action out of the pipeline or convert all pipeline to a for-loop. It's better to avoid operations with side effects in stream pipelines.